iib/workers/tasks/legacy.py

```diff
@@ -0,0 +1,226 @@
+
+
+def get_legacy_support_packages(bundles):
+    """
+    Checks if legacy support is required for any packages in the bundles.
+
+    :param list<str> bundles: a list of strings representing the pull specifications of the bundles
+        to add to the index image being built.
+    :return: a set of packages that require legacy support
+    :rtype: set
+    """
+    packages = set()
+    for bundle in bundles:
+        skopeo_out = skopeo_inspect(f'docker://{bundle}')
+        if skopeo_out['Labels'].get('com.redhat.delivery.backport', False):
+            packages.add(skopeo_out['Labels']['operators.operatorframework.io.bundle.package.v1'])
+
+    return packages
+
+
+def _podman_login():
+    """
+    Login into the registry to be able to access private repos.
+
+    :raises iib.exceptions.IIBError: if the login fails
+    """
+    conf = get_worker_config()
+    log.info('Logging in %s using podman', conf['iib_registry'])
+    username, password = conf['iib_registry_credentials'].split(':')
+    run_cmd(
+        ['podman', 'login', '-u', username, '-p', password, conf['iib_registry']],
+        exc_msg=f'Failed to login into the registry {conf["iib_registry"]}',
+    )
+
+
+def _verify_package_info(package_dir, from_index):
+    """
+    Verifies if the exported package info is generated correctly
+
+    :param str package_dir: path to the exported package directory
+    :param str from_index: the pull specification of the image image
+    :raises iib.exceptions.IIBError: if the generated package info is missing
+    """
+    package_name = package_dir.split('/download-', 1)[-1]
+    log.info('Verifying package_name %s', package_name)
+    # opm does not fail when the package is missing in the index image, hence we
+    # check the number of generated files. If it's equal to 1, that means only an empty
+    # `package.yaml` file is generated and the package is missing
+    if len(os.listdir(package_dir)) == 1:
+        raise IIBError(f'package {package_name} is missing in index image {from_index}')
+
+    try:
+        output = api.build_and_verify(source_dir=package_dir)
+        if not output.is_valid:
+            raise IIBError(
+                f'The exported package {package_name} is invalid: '
+                f'{", ".join(output.validation_dict["errors"])}'
+            )
+    except errors.OpCourierBadBundle as err:
+        raise IIBError(f'The exported package {package_name} is invalid: {err}')
+
+
+def _zip_package(package_dir):
+    """
+    Zip content of exported package to a `manifests.zip` file
+
+    :param str package_dir: path to the exported package directory
+    :raises iib.exceptions.IIBError: if unable to zip the exported package
+    """
+    base_dir, package_name = package_dir.split('/download-', 1)
+    try:
+        shutil.make_archive(f'{base_dir}/manifests', 'zip', package_dir)
+    except Exception as err:
+        raise IIBError(f'Unable to zip exported package for {package_name}: {err}')
```

Comment:
Perform a `log.exception` here so that we can see the traceback, but not expose the gnarly error to the user.